.github/workflows/security.yml

```diff
@@ -0,0 +1,111 @@
+name: Security Scan
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  trivy:
+    name: Trivy scan
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: CheckouSecurityt code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Run Trivy scan
+        id: trivy
+        uses: aquasecurity/trivy-action@master
+        continue-on-error: true
+        with:
+          scan-type: 'fs'
+          scanners: 'vuln,secret'
+          severity: 'CRITICAL,HIGH'
+          ignore-unfixed: true
+          format: 'template'
+          template: '@.github/workflows/trivy-template.md.tpl'
+          output: 'trivy-report.md'
+          timeout: '5m'
+
+      - name: Comment Trivy scan on PR
+        if: github.event_name == 'pull_request'
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          path: trivy-report.md
+          header: trivy-scan
+
+      - name: Fail if Trivy found high/critical issues
+        if: steps.trivy.outcome != 'success'
+        run: |
+          echo "High or critical vulnerabilities found by Trivy."
+          exit 1
+
+  security-scan:
+    name: Gosec scan
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+
+      - name: Install gosec
+        run: go install github.com/securego/gosec/v2/cmd/gosec@latest
```

Comment:
It's the first time I see this `gosec` tool but it seems they make it easy to integrate with GH actions, see https://github.com/securego/gosec?tab=readme-ov-file#github-action